variants/arduino_101/linker_scripts/flash.ld

```diff
@@ -158,6 +158,9 @@ SECTIONS
     {
         . = ALIGN(4);
         __start_heap = .;
+        end = __start_heap;
+        _end = end;
+        __end = end;
```

Comment:
This is the first step to properly support newlib's `nano.specs`, which unfortunately aren't compiled in the current toolchain. The nano lib doesn't provide all the symbols and the linker would fail with something like 
```
sbrk.c:(.text._sbrk+0x2): undefined reference to `end'
```